src/ant/include/ant/AntennaChecker.hh

```diff
@@ -217,6 +279,65 @@ class AntennaChecker
   std::string report_file_name_;
 
   static constexpr int max_diode_count_per_gate = 10;
+
+  /////////////////////////////////////////////////////////////////////////////////
+  std::unordered_map<odb::dbTechLayer*, GraphNodeVector> node_by_layer_map_;
+  std::unordered_map<std::string, LayerInfoVector> info_;
+  std::vector<Violation> antenna_violations_;
+  int node_count_;
+  dbTechLayer* min_layer_;
+  // dsu variables
+  std::vector<int> dsu_parent_, dsu_size_;
+
+  // DSU functions
+  void init_dsu();
+  int find_set(int v);
+  void union_set(int u, int v);
+  bool dsu_same(int u, int v);
+
+  bool isValidGate(dbMTerm* mterm);
+  void buildLayerMaps(dbNet* net);
+  void checkNet(dbNet* net,
+                bool verbose,
+                bool report,
+                std::ofstream& report_file,
+                dbMTerm* diode_mterm,
+                float ratio_margin,
+                int& net_violation_count,
+                int& pin_violation_count);
+  void saveGates(dbNet* db_net);
+  void calculateAreas();
+  void calculatePAR();
+  void calculateCAR();
+  int checkInfo(dbNet* db_net,
```

Comment:
Change `checkInfo` for a more meaningful name (related to the comments about InfoType being a bad struct name).